raiden/raiden_service.py

```diff
@@ -1022,7 +1021,7 @@ def mediated_transfer_async(
         target: TargetAddress,
         identifier: PaymentID,
         fee: FeeAmount = MEDIATION_FEE,
-        secret: Secret = None,
+        secret: Optional[Secret] = None,
         secrethash: SecretHash = None,
```

Comment:
I'll remove `secrethash` here.  Anyway `secrethash` is computed from `secret` later in `start_mediated_transfer_with_secret()`.

(First I want to see the tests pass.)

Reply:
But no, that would change the API.

Reply:
Actually, there is a use case with only the secrethash but not the secret.